### PR TITLE
Add default installation locations for more C compilers

### PR DESCRIPTION
**Pull request by tomrittervg**
This converts the file to the new config format (AIUI) and adds more C
compilers commonly provided by package managers.

### CODE REVIEW

File: etc/config/c.defaults.properties [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
 objdumper=objdump
+postProcess=
 supportsBinary=true
 binaryHideFuncRe=^(__.*|_(init|start|fini)|(de)?register_tm_clones|call_gmon_start|frame_dummy|\.plt.*)$
 stubRe=\bmain\b
 stubText=int main(void){return 0;/*stub provided by Compiler Explorer*/}
 supportsLibraryCodeFilter=true
+
+group.gcc.compilers=g44:g45:g46:g47:g48:g5:g6:g7:g8:gdefault
+compiler.g44.exe=/usr/bin/gcc-4.4
```
Comment (partouf) [inline etc/config/c.defaults.properties:14]: Compilers need to have unique names (for the API). We usually put a c in front of the name here, so cg44 in this case.

**Review by partouf — CHANGES_REQUESTED** [review]

File: etc/config/c.defaults.properties (hunk 1 above)
Comment (tomrittervg) [reply etc/config/c.defaults.properties]: Updated!

**Comment by tomrittervg** [comment]
Updated with cg44 and such. FWIW `make check` did not alert me to this so it might be worth adding such a check there but I'm not sure how to do so.

**Comment by partouf** [comment]
> Updated with cg44 and such. FWIW `make check` did not alert me to this so it might be worth adding such a check there but I'm not sure how to do so.

None of the unittests check the actual configuration, just the code. I think we have some checks during startup, but yes, might be worth looking into this a bit more.

**Review by partouf — APPROVED** [review]

**Comment by partouf** [comment]
Thanks!

**Pull request merged**